src/Slim/Middleware/HandlerTrait.php

```diff
@@ -0,0 +1,23 @@
+<?php
```

Comment:
Do we really need a trait? I think it's not necessary neither valid.

Reply:
I can move those codes to ```Session.php```.
PS: I do not like the idea auto-registered via autoload.files, what if user do not want to use session middleware as global? they will think about why they should call ```session_set_save_handler(...)``` everytime when Composer autoloader started.

Reply:
`session_set_save_handler` registers handler for application, not just middleware.
So, it's same as putting it in a file and adding `autoload.files` to their own `composer.json`.

Reply:
About it, somtimes ```session_set_save_handler``` registers handler for application, true. but for example:
``` php
//User do want session start
$route->group('/web' ,fn() {...})->add(SessionMiddleware::class);
//User think session is unnecessary
$route->group('/api', fn() {...});
```
Thas is the main reason I do not agree with ```session_set_save_handler(...)``` auto-called via Composer autoload scripts

If ```session_set_save_handler(...)``` is totally controlled by other packages, I think user will confused when some packages do not called ```auto-regist``` every launching of APP, this PR would be unnecessary.

Reply:
I understand what you say. But handler is registered for entire script (not even app, group...).

If you want, try that code, and handler registered at `/web` is same used when requesting `/api`, because of the way `add($class)` works.

Reply:
OK, so I tested your sample, and in fact it works like you said. 👍 
Only missing thing is comment below, about where to set the handler.

Reply:
@bryanjhv Check those codes plz:

 [```DeferredCallable```](https://github.com/slimphp/Slim/blob/3.x/Slim/DeferredCallable.php) and [```MiddlewareAwareTrait::addMiddleware(...)```](https://github.com/slimphp/Slim/blob/3.x/Slim/MiddlewareAwareTrait.php)
```add(...)``` is lazy-loading impl, not equal invoked immediately.It is just invoked when a new request come which route has resisted middlware matched.

So,why I should require a script and call ```session_set_save_handler(...)``` when the coming request would not need start session? And what if user has his own SessionHandlerInterface impl? Let him write a psr-4 autoload file in composer.json? Come on man, it is not a good idea.

I wrote some code for the talk:

>  handler is registered for entire script (not even app, group, route, whatever).

> If you want, try that code, and handler registered at /web is same used when requesting /api  (no matter if you use it or not), because of the way add($class) works.

 you can run it and test it:
```php
<?php
require __DIR__.'/vendor/autoload.php';

class M 
{
    public function __invoke()
    {
        printf("%s::class have been invoked\n", get_class());
        die;
    }
}

$app = new Slim\App();
$app->get('/web', function ($req, $res, $args) {})->add(M::class);
$app->get('/api', function ($req, $res, $args) {});

$app->run();
```
The best thing is I would never seen the HTTP response body like "M::class have been invoked" when I send a request to ```/api```

Reply:
FAQ: Where to set the handler?

@bryanjhv You can see it when you review the [USE CASE](https://github.com/hidehalo/slim-session/blob/master/tests/SessionTest.php) about ```SessionTest::testStartSession(...)```

I should get sleep now, I will push some new commits if you like it. GN

Reply:
Done it